mulog-slack/src/com/brunobonacci/mulog/publishers/slack.clj

```diff
@@ -0,0 +1,118 @@
+(ns com.brunobonacci.mulog.publishers.slack
+  (:require [com.brunobonacci.mulog.publisher :as p]
+            [com.brunobonacci.mulog.buffer :as rb]
+            [com.brunobonacci.mulog.utils :as ut]
+            [clj-http.client :as http]
+            [cheshire.core :as json]
+            [clj-time.format :as tf]
+            [clj-time.coerce :as tc]
+            [clojure.string :as str]))
+
+
+(defn- unix-ms-to-iso8601
+  [unix-ms]
+  (let [iso-8601-fmt (tf/formatters :date-time)]
+    (->> unix-ms
+        (tc/from-long)
+        (tf/unparse iso-8601-fmt))))
+
+
+(defn- default-render-message
+  "Timestamp and event name with the log content in a code block"
+  [event]
+  (let [timestamp (unix-ms-to-iso8601 (:mulog/timestamp event))
+        event-name (:mulog/event-name event)]
+    (str timestamp " - " event-name
+        \newline
+        "```"
+        \newline
+        (ut/pprint-event-str event)
+        "```")))
+
+(defn- send-slack-message
+  [webhook-url message publish-delay]
+  ;; (println (str "About to send: " message " to " webhook-url))
+  (http/post
+   webhook-url
+   {:content-type "application/json"
+    :accept :json
+    ;; TODO: How is publish-delay to be implemented?
+    ;; :socket-timeout publish-delay
+    ;; :connection-timeout publish-delay
```

Comment:
you can set the timeout to 3s (`3000`)

Reply:
Ah, I misunderstood. I thought this is how the the repeated publishing is done.
The call to `rb/recurring-task` in core is actually what does it.
The use of `publish-delay` here is just so that if there's an incoming publish, if there's a pending publish, then the pending publish times out.